lib/internal/test_runner/runner.js

```diff
@@ -241,22 +241,29 @@ function runTestFile(path, root, inspectPort, filesWatcher) {
       err = error;
     });
 
-    if (isUsingInspector()) {
-      const rl = createInterface({ input: child.stderr });
-      rl.on('line', (line) => {
-        if (isInspectorMessage(line)) {
-          process.stderr.write(line + '\n');
-        }
-      });
-    }
-
-    const parser = new TapParser();
-    child.stderr.pipe(parser).on('data', (ast) => {
-      if (ast.lexeme && isInspectorMessage(ast.lexeme)) {
-        process.stderr.write(ast.lexeme + '\n');
+    const rl = createInterface({ input: child.stderr });
+    rl.on('line', (line) => {
+      if (isInspectorMessage(line)) {
+        process.stderr.write(line + '\n');
+        return;
       }
+
+      // stderr cannot be treated as TAP, per the spec. However, we want to
+      // surface stderr lines as TAP diagnostics to improve the DX. Inject
+      // each line into the test output as an unknown token as if it came
+      // from the TAP parser.
+      const node = {
+        kind: TokenKind.UNKNOWN,
+        node: {
+          value: line,
+        },
+      };
+
+      subtest.addToReport(node);
```

Comment:
this will cause the message to print on stdout, why not write it to stderr?

Reply:
i.e. just remove the `isInspectorMessage` condition

Reply:
I did it this way for two reasons:

- It makes the logs on stderr look more consistent with those from stdout.
- More importantly, it keeps the logs grouped together better. For example, the CLI runner spawn N child processes. If they all start logging to stderr, those logs will be mixed together. By associating those logs with the child process output, the ordering of the logs is a bit better.

Reply:
I wonder how that can affect tools looking for output on stderr (debuggers, for example). I guess we can ship this and change if there is any feedback